socorro/lib/util.py

```diff
@@ -144,3 +146,30 @@ def _retry_fun(*args, **kwargs):
         return _retry_fun
 
     return _retry_inner
+
+
+# Bug 1911612: Temporary decorator to log the method name, arguments and traceback
+# for each ESCrashStorage and SuperSearch method to better understand where ES gets
+# used by the webapp.
+def es_usage_logger(func):
+    @wraps(func)
+    def wrapper(self, *args, **kwargs):
+        stack_trace = "".join(traceback.format_stack())
+        class_name = self.__class__.__name__
+        method_name = func.__name__
+
+        log_entry_part_one = {
+            "bug_id": "1911612",
+            "method": f"{class_name}.{method_name}",
+            "arguments": {"args": args, "kwargs": kwargs},
+        }
+
+        log_entry_part_two = f"\ntraceback:\n{stack_trace}"
+
+        self.logger.info(json.dumps(log_entry_part_one, indent=4) + log_entry_part_two)
```

Comment:
This is very verbose right now, thanks largely to the traceback. I'm a little concerned about increases in log volume, even if temporary.

For example, a single call to `SuperSearchUnredacted.get_implementation` had 154 lines where all but 10 were the traceback.

Here's a snippet of the first chunk as a reference:
```
webapp-1         | 2024-09-12 21:16:22,736 INFO - webapp - crashstats.supersearch.models.SuperSearchUnredacted - {
webapp-1         |     "bug_id": "1911612",
webapp-1         |     "method": "SuperSearchUnredacted.get_implementation",
webapp-1         |     "arguments": {
webapp-1         |         "args": [],
webapp-1         |         "kwargs": {}
webapp-1         |     }
webapp-1         | }
webapp-1         | traceback:
webapp-1         |   File "/usr/local/lib/python3.11/threading.py", line 1002, in _bootstrap
webapp-1         |     self._bootstrap_inner()
webapp-1         |   File "/usr/local/lib/python3.11/threading.py", line 1045, in _bootstrap_inner
...
```

I'd like some way to reduce the log volume without sacrificing signal. Sampling is one option, but that may mean missing some little used code paths. A lot of the traceback is framework/middleware code, so another option could be to e.g. remove lines that start with `'File "/usr/local/'` and the next line after and lines that contain the substring "middleware".

Is this additional log volume a concern? Is there a good way to estimate the temporary increase? Where might I find log volume costs/limits in GCP? Maybe it's okay as-is particularly if the increase is timeboxed to say a month.